testng-eclipse-plugin/src/main/org/testng/eclipse/util/param/TestNGMethodParameterVisitor.java

```diff
@@ -165,16 +169,41 @@ protected boolean isKnownAnnotation(String annotationType) {
   }
 
   /**
-   * @return
+   * @return the parameter name/value map
    */
   public Map<String, String> getParametersMap() {
     Map<String, String> parameterMap = Maps.newHashMap();
-    for (List<String> paramNames : m_parameters.values()) {
-      for(int i= 0; i < paramNames.size(); i++) { 
-        parameterMap.put(paramNames.get(i), "not-found");
+    for (Entry<MethodDeclaration, List<String>> paramEntry : m_parameters.entrySet()) {
+      // get all '@Optional' value, as the default value for the parameters
+      List<String> optionals = Lists.newArrayList();
+      for (Object p : paramEntry.getKey().parameters()) {
+        SingleVariableDeclaration paramDecl = (SingleVariableDeclaration) p;
+        for (Object pm : paramDecl.modifiers()) {
+          SingleMemberAnnotation paramAnn = (SingleMemberAnnotation) pm;
+          if ("Optional".equals(paramAnn.getTypeName().toString())) {
+            optionals.add(paramAnn.getValue().toString());
+          }
+        };
+      };
+
+      List<String> paramNames = paramEntry.getValue();
+      for(int i = 0; i < paramNames.size(); i++) {
+        String val = DEFAULT_PARAM_VALUE;

```

Comment:
however, since Parameters are test level (not each test method level), so if there are parameters for each test method, the last will win (be put in the parameters map)